elytron/WFLY-20384-properties-role-mapper.adoc

```diff
@@ -0,0 +1,219 @@
+:email:             pesilva@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Some people migrating from previous versions of WildFly to the latest version are struggling with the migration of the RoleMapping login module. The equivalent of the RoleMapping login module is the mapped-role-mapper, which can achieve similar behavior.
+
+The main challenge comes from the fact that a properties file was previously used to map REALM roles to APPLICATION roles, and with mapped-role-mapper, this no longer seems possible. The proposal is to create a new role mapper based on a properties file to match REALM roles to APPLICATION roles.
+
+=== User Stories
+
+Some users need to map **REALM** roles to **APPLICATION** roles. The example https://github.com/gabrielpadilh4/custom-properties-mapped-role-mapper[custom-properties-mapped-role-mapper] was created to avoid this gap.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-20384[WFLY-20384]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/ELY-2892[ELY-2892]
+* https://issues.redhat.com/browse/WFCORE-7203[WFCORE-7203]
+* https://issues.redhat.com/browse/WFLY-20522[WFLY-20522]
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly-security/wildfly-elytron[Elytron]
+* https://github.com/wildfly/wildfly-core/[Wildfly Core - Elytron]
+* https://github.com/wildfly/wildfly[Wildfly Documentation]
+
+=== Other Interested Projects
+N/A
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [x] Managed domain
+* [x] OpenShift Source-to-Image (S2I)
+* [ ] Bootable jar
+
+== Requirements
+
+This RFE will add an attribute to add a new elytron subsystem role mapper configuration to map roles based on a properties file. The CLI command should have two paramenters: `path` and `relative-to`. For example:
+
+```
+/subsystem=elytron/properties-role-mapper=prm:add(path="my-role-properties.properties", relative-to="jboss.server.config.dir")
+```
+
+* The `path` paramenter is mandary, and represents the properties file;
+* The `relative-to` is not mandadoty, represents the relative directory where the file is present. Acepts environment variables, like: `jboss.server.config.dir`
+
+It's possible use the full path like `path="eap/stadanloe/configuration/my-role-properties.properties`, in this case the `relative-to` paramenter is not necessary, however, if `relative-to` is used, the `path` paramenter should be provided.
+
+The properties file should contain the the Realm role, followed by the Application mapped role, for example:
+
+```
+$ cat my-role-properties.properties
+
+EnterpriseRole=role-admin,role-123
+```
+
+Based on that file, the new `PropertiesMappedRoleMapper.java` will map the roles.
+
+=== Changed requirements
+Update the schema n order to add the new `properties-role-mapper` configuration.
+
+=== Non-Requirements
```

Comment:
IIUIC, the original login module is able to load also property files from deployments: https://issues.redhat.com/browse/WFCORE-7203 I assume such functionality is not planned.

Reply:
No managed components should be loading from deployed content.

Reply:
Also, the original module has a `replaceRole` property. Also out of the scope?